skimage/segmentation/tests/test_watershed.py

```diff
@@ -392,7 +392,16 @@ def test_watershed12(self):
         for lab, area in zip(range(4), [34,74,74,74]):
             self.assertTrue(np.sum(ws == lab) == area)
 
+    def test_watershed13(self):
+        """Test on markers
 
+        This is here to ensure input markers is not modified
+        """
+        image = np.zeros((21, 21))
+        markers = np.zeros((21, 21), int)
+        original_markers = np.copy(markers)
+        watershed(image, markers)
+        np.testing.assert_equal(original_markers, markers)
```

Comment:
Ah, a couple of comments on this test @decorouz:

1. Please rename the test, `test_watershed_input_not_modified`. (I see that there's other poorly-named tests above it, but today is the best day to start improving our test names! 😃)
2. Unfortunately, in this situation the test is not going to guarantee correctness, because there are no markers (markers have value >0), so watershed will do nothing. I suggest instead

```python
image = np.random.default_rng().random(size=(21, 21))
markers = np.zeros((21, 21), dtype=np.uint8)
markers[[5, 5, 15, 15], [5, 15, 5, 15]] = [1, 2, 3, 4]
original_markers = np.copy(markers)
result = watershed(image, markers)
np.testing.assert_equal(original_markers, markers)
assert not np.all(result == markers)
```

The last guard will prevent the issue I raised above from reappearing.

I think this is really the last thing! 😅 Thank you! 🙏

Reply:
(Note: in case you're unfamiliar with NumPy array fancy indexing on line 3:

https://numpy.org/doc/stable/user/basics.indexing.html#integer-array-indexing
)

Reply:
I did consider naming the test differently, however since the other tests in the class followed a similar pattern, I decided to follow the same. Your suggestion is better. And yes, It's test names improvement day 😃. 

Thank you for your reviews and contributions. I hope it is the last thing for this issue. I enjoyed it.